src/main/java/util/Maths.java

```diff
@@ -0,0 +1,53 @@
+package util;
+
+public class Maths {
+	public static double clamp(double x, double min, double max) {
+		if (x < min) return min;
+		if (x > max) return max;
+		return x;
+	}
+
+	// See: https://www.desmos.com/calculator/3nakwtzuis
+	public static double sigmoid(double x, double fact, double offset) {
+		return 1 / (1 + Math.exp(-fact*x - offset));
+	}
+
+	public static double max(double[] xs) {
```

Comment:
This is a nice example regarding our "reinventing the wheel" discussion. Math.max definitely exists already and I don't see why we should re-implement it. Also, your min function is never used.

Reply:
I didn't actually see an overload of Math.max that takes a list as input. Otherwise we can gladly use that of course.
And you're right, the min isn't used, we can remove that too

Reply:
But I think the Math.max function cannot handle arrays and since we are dealing with potentially big arrays, I think it is more efficient to use that simple if-statement in line 18, instead of having a method-call to Math.max each time

Reply:
Oh wow, I didn't actually read that this is an array max, forget I ever said anything

Reply:
no worries xD